runtime/vm/visible.c

```diff
@@ -64,7 +64,6 @@ checkVisibility(J9VMThread *currentThread, J9Class* sourceClass, J9Class* destCl
 				&& (J2SE_VERSION(vm) >= J2SE_19) 
 				&& J9_ARE_ALL_BITS_SET(vm->runtimeFlags, J9_RUNTIME_JAVA_BASE_MODULE_CREATED)
 				&& !J9ROMCLASS_IS_PRIMITIVE_TYPE(destClass->romClass)
-				&& J9_ARE_ALL_BITS_SET(vm->runtimeFlags, J9_RUNTIME_DENY_ILLEGAL_ACCESS)
```

Comment:
I'm not sure I follow why this doesn't need to be checked as part of the visibility check.  Can you elaborate?

Reply:
It is still being checked as part of the visibility check but just at a later stage. We check it further down when `isPackageExportedToModuleWithName(...)` is called.

The --illegal-access=[permit|debug|warn] opens all packages to all unnamed modules. So it is similar to add-exports but allows deeper reflection. The 'deeper reflection' part is handled by the JCL but we still have to make those packages visible in the vm.